introducing idefics (#1386)
huggingface/blog: introducing-idefics.md

---
title: "Introducing IDEFICS: An Open Reproduction of State-of-the-art Visual Langage Model"
thumbnail: /blog/assets/71_ethical-charter/thumbnail.jpg

> authors:

Reviewer: For guests, you can add `guest: true` as in https://github.com/huggingface/blog/blob/main/introducing-doi.md?plain=1#L9

- user: HugoLaurencon
- user: davanstrien
- user: stas
- user: Leyo
- user: SaulLu
- user: TimeRobber
- user: skaramcheti
- user: aps
- user: giadap
- user: VictorSanh
---

# Introducing IDEFICS: An Open Reproduction of State-of-the-art Visual Langage Model

<!-- {blog_metadata} -->
<!-- {authors} -->

<!-- TODO: recognize full list of co-authors -->

> We are excited to release IDEFICS (**I**mage-aware **D**ecoder **E**nhanced à la **F**lamingo with **I**nterleaved **C**ross-attention**S**), an open-access visual language model. IDEFICS is based on [Flamingo](https://huggingface.co/papers/2204.14198), a state-of-the-art visual language model initially developed by DeepMind, which has not been released publicly. Similarly to GPT-4, the model accepts arbitrary sequences of image and text inputs and produces text outputs. IDEFICS is built solely on publicly available data and models and comes in two variants: the base version and the instructed version.

Reviewer: Maybe mention the model size around the end

> We are excited to release IDEFICS (**I**mage-aware **D**ecoder **E**nhanced à la **F**lamingo with **I**nterleaved **C**ross-attention**S**), an open-access visual language model. IDEFICS is based on [Flamingo](https://huggingface.co/papers/2204.14198), a state-of-the-art visual language model initially developed by DeepMind, which has not been released publicly. Similarly to GPT-4, the model accepts arbitrary sequences of image and text inputs and produces text outputs. IDEFICS is built solely on publicly available data and models and comes in two variants: the base version and the instructed version.

Reviewer: Mention somewhere that this is an open access, llama 1-based model

> We believe that the development of state-of-the-art AI models should be more transparent. Our goal with IDEFICS is to reproduce and provide the AI community with systems that match the capabilities of large proprietary models like Flamingo. As such, we took important steps contributing to bringing transparency to these AI systems: we used only publicly available data, we provided tooling to explore training datasets, we shared [technical lessons and mistakes](https://github.com/huggingface/m4-logs/blob/master/memos/README.md) of building such artifacts and assessed the model’s harmfulness by adversarially prompting it before releasing it. We are hopeful that IDEFICS will serve as a solid foundation for more open research in multimodal AI systems.

Reviewer: maybe this would sound more confident
```suggestion
The development of state-of-the-art AI models should be more transparent. Our goal with IDEFICS is to reproduce and provide the AI community with systems that match the capabilities of large proprietary models like Flamingo. As such, we took important steps contributing to bringing transparency to these AI systems: we used only publicly available data, we provided tooling to explore training datasets, we shared [technical lessons and mistakes](https://github.com/huggingface/m4-logs/blob/master/memos/README.md) of building such artifacts and assessed the model’s harmfulness by adversarially prompting it before releasing it. We are hopeful that IDEFICS will serve as a solid foundation for more open research in multimodal AI systems.
```

> We believe that the development of state-of-the-art AI models should be more transparent. Our goal with IDEFICS is to reproduce and provide the AI community with systems that match the capabilities of large proprietary models like Flamingo. As such, we took important steps contributing to bringing transparency to these AI systems: we used only publicly available data, we provided tooling to explore training datasets, we shared [technical lessons and mistakes](https://github.com/huggingface/m4-logs/blob/master/memos/README.md) of building such artifacts and assessed the model’s harmfulness by adversarially prompting it before releasing it. We are hopeful that IDEFICS will serve as a solid foundation for more open research in multimodal AI systems.

Reviewer: ```suggestion
The development of state-of-the-art AI models should be more transparent. Our goal with IDEFICS is to reproduce and provide the AI community with systems that match the capabilities of large proprietary models like Flamingo. As such, we took important steps contributing to bringing transparency to these AI systems:

* We used only publicly available data.
* We provided tooling to explore training datasets.
* We shared [technical lessons and mistakes](https://github.com/huggingface/m4-logs/blob/master/memos/README.md) of building such artifacts and assessed the model’s harmfulness by adversarially prompting it before releasing it.
```

> We believe that the development of state-of-the-art AI models should be more transparent. Our goal with IDEFICS is to reproduce and provide the AI community with systems that match the capabilities of large proprietary models like Flamingo. As such, we took important steps contributing to bringing transparency to these AI systems: we used only publicly available data, we provided tooling to explore training datasets, we shared [technical lessons and mistakes](https://github.com/huggingface/m4-logs/blob/master/memos/README.md) of building such artifacts and assessed the model’s harmfulness by adversarially prompting it before releasing it. We are hopeful that IDEFICS will serve as a solid foundation for more open research in multimodal AI systems.

Reviewer 2: would it make sense to cite (and link to, for instance if artefacts are on the Hub) other models that have already been shared by the community? (i'm no expert but OpenFlamingo comes to mind?)

Reviewer 2: also maybe worth mentioning that openai does _not_ have something similar shipped (idk if relevant, for me it's an interesting "selling" point)

> Try out the [demo](TODO: link) and the [models](https://huggingface.co/HuggingFaceM4/idefics-80b-instruct) on the hub!

Reviewer: ```suggestion
Try out the [demo](TODO: link) and the [models](https://huggingface.co/HuggingFaceM4/idefics-80b-instruct) on the Hub!
```

![Screenshot of IDEFICS generation for HF Woodstock of AI](TODO)

## What is IDEFICS?

> IDEFICS is an 80 billion parameters multimodal model that accepts sequences of images and texts as input, and generates coherent text as output. It can answer questions about images, describe visual content, create stories grounded in multiple images, etc.

Reviewer: ```suggestion
IDEFICS is an 80 billion parameters multimodal model that accepts sequences of images and texts as input and generates coherent text as output. It can answer questions about images, describe visual content, create stories grounded in multiple images, etc.
```

IDEFICS is an open-access reproduction of Flamingo and is comparable in performance with the original closed-source model across various image-text understanding benchmarks. It comes in two variants - 80 billion parameters and 9 billion parameters.

![Plot comparing the performance of Flamingo, OpenFlamingo and IDEFICS](TODO)

We also provide fine-tuned versions [idefics-80B-instruct](https://huggingface.co/HuggingFaceM4/idefics-80b-instruct) and [idefics-9B-instruct](https://huggingface.co/HuggingFaceM4/idefics-9b-instruct) adapted for conversational use cases.

## Training Data

> IDEFICS was trained on a mixture of openly available datasets: Wikipedia, Public Multimodal Dataset and LAION, as well as a new 115B token dataset called [OBELICS](https://huggingface.co/datasets/HuggingFaceM4/OBELICS) that we created. OBELICS consists of 141 million interleaved image-text documents scraped from the web and contains 353 million images.

Reviewer: ```suggestion
IDEFICS was trained on a mixture of openly available datasets: Wikipedia, Public Multimodal Dataset, and LAION, as well as a new 115B token dataset called [OBELICS](https://huggingface.co/datasets/HuggingFaceM4/OBELICS) that we created. OBELICS consists of 141 million interleaved image-text documents scraped from the web and contains 353 million images.
```

## Ethical evaluation

At the outset of this project, through a set of discussions, we developed an ethical charter (https://huggingface.co/blog/ethical-charter-multimodal) that would help steer decisions made during the project. This charter sets out values, including being self-critical, transparent, and fairness which we have sought to pursue in how we approached the project itself and the release of this model.

As part of the release process, we internally evaluated the model for potential biases by adversarially prompting the model with images and text that might elicit responses we do not want from the model (a process known as red teaming). We also carried out a more systematic evaluation of potential biases in the model.

> Please try out IDEFICS with the [demo](TODO) and let us know your feedback by using the community tab! We are committed to improving these models and making large multimodal AI models more accessible to the machine learing community.

Reviewer: ```suggestion
Please try out IDEFICS with the [demo](TODO) and let us know your feedback using the community tab! We are committed to improving these models and making large multimodal AI models accessible to the machine learning community.
```

## Getting Started with IDEFICS

> IDEFICS models are available on the Hugging Face Hub. Here is code sample to try it out:

Reviewer: ```suggestion
IDEFICS models are available on the Hugging Face Hub and supported in the last `transformers` version. Here is a code sample to try it out:
```
